data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsPublisher.java

```diff
@@ -26,22 +28,23 @@
 public class MetricsPublisher {
 
   private static final Logger LOG = LogManager.getLogger();
+  private static final MediaType MEDIA_TYPE = MediaType.parse("application/json; charset=utf-8");
   private final OkHttpClient client;
-  private final MediaType mediaType = MediaType.parse("application/json; charset=utf-8");
+  private final HttpUrl endpoint;
 
-  public MetricsPublisher(final OkHttpClient client) {
+  public MetricsPublisher(final OkHttpClient client, final HttpUrl endpoint) {
     this.client = client;
+    this.endpoint = endpoint;
   }
 
-  public int publishMetrics(final String endpointAddress, final String json) throws IOException {
-    HttpUrl endpoint = HttpUrl.get(endpointAddress);
-    RequestBody body = RequestBody.create(json, mediaType);
+  public int publishMetrics(final String json) throws IOException {
+    RequestBody body = RequestBody.create(json, MEDIA_TYPE);
     Request request = new Request.Builder().url(endpoint).post(body).build();
     Response response = this.client.newCall(request).execute();
     int responseCode = response.code();
     response.close();
-    if (responseCode != 200) {
-      LOG.error(
+    if (responseCode != SC_OK) {
+      LOG.debug(
           "Failed to publish metrics to external metrics service. Response code {}", responseCode);
     }
     return responseCode;
```

Comment:
I don't think this response code is actually being used (except possibly in tests which probably doesn't need to happen)